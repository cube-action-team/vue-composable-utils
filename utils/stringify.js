const stringifyObject = (object, brace = ';') => {
  const keyValuePairs = Object.entries(object).filter(pair => pair[0] || pair[1]);

  return keyValuePairs.reduce((acc, item, index) => {
    if (index === keyValuePairs.length - 1) {
      return (acc = acc + `${item[0]}=${item[1]}`);
    }

    return (acc = acc + `${item[0]}=${item[1]}${brace}`);
  }, '');
};

export { stringifyObject };
