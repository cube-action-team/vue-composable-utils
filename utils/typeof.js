export const isClient = typeof window !== 'undefined';

export const isArray = Array.isArray;

export const isFunction = val => typeof val === 'function';

export const isString = val => typeof val === 'string';

export const isSymbol = val => typeof val === 'symbol';

export const isBoolean = val => typeof val === 'boolean';

export const isDate = val => isObject(val) && isFunction(val.getTime);

export const isNumber = val => typeof val === 'number';

export const isObject = val => val !== null && typeof val === 'object';

export const isElement = val => isObject(val) && !!val.tagName;

export function isPromise(val) {
  return isObject(val) && isFunction(val.then) && isFunction(val.catch);
}
