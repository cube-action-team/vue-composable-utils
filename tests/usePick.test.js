import { usePick } from '@/';
import { mount, createLocalVue } from '@vue/test-utils';
import { mountComposition } from './utils';
import VueCompositionAPI from '@vue/composition-api';

const localVue = createLocalVue();
localVue.use(VueCompositionAPI);

const obj = { a: 1, b: 2, c: 3, d: 4 };
const mock = { a: 1, d: 4 };
let arr = ['a', 'd'];

const PickComponent = localVue.component('pick-component', {
  setup() {
    const pick = usePick(obj, arr);

    return {
      pick,
    };
  },
  template: `
    <div>
    <p>Pick: {{ JSON.stringify(pick) }}</p>
  </div>
        `,
});

describe('usePick', () => {
  it('should be defined useState', () => {
    expect(usePick).toBeDefined();
  });

  it('should get pick value', async () => {
    const wrapper = mount(PickComponent);
    expect(wrapper.vm.pick).toEqual(mock);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should get count value when trigger method', async () => {
    arr = [];
    const wrapper = mount(
      mountComposition(() => {
        const pick = usePick(obj, arr);

        return {
          pick,
        };
      }),
      { localVue },
    );

    expect(wrapper.vm.pick).toEqual({});
  });

  it('should get pick object empty', async () => {
    const wrapper = mount(
      mountComposition(() => {
        const pick = usePick(obj);

        return {
          pick,
        };
      }),
      { localVue },
    );

    expect(wrapper.vm.pick).toEqual({});
  });

  it('should get pick object array', async () => {
    const wrapper = mount(
      mountComposition(() => {
        const pick = usePick([]);

        return {
          pick,
        };
      }),
      { localVue },
    );

    expect(wrapper.vm.pick).toEqual([]);
  });

  it('should get pick object equal one', async () => {
    const wrapper = mount(
      mountComposition(() => {
        const pick = usePick(obj, { a: 'b' });

        return {
          pick,
        };
      }),
      { localVue },
    );

    expect(wrapper.vm.pick).toEqual({ b: 1 });
  });
});
