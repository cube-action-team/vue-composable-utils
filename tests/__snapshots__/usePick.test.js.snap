// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePick should get pick value 1`] = `
<div>
  <p>Pick: {"a":1,"d":4}</p>
</div>
`;
